### PR TITLE
adding a method for fetching failure reason from module

### PR DESCRIPTION
**Pull request by MoodyMudit**
Adding a method in modules that can tell the audit runner what it was doing. This message can be then clubbed with failure reason from comparator to tell the user what actually failed on the system.

### CODE REVIEW

File: hubblestack/audit/command_line_parser.py [hunk 1]
```diff
@@ -457,3 +457,22 @@ def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
...
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': {'cmdline': 'app --config-file=test test'}, 'status': True},
+                  'caller': 'Audit'}
+    :return:
+    """
+    key_aliases = runner_utils.get_param_for_module(block_id, block_dict, 'key_aliases')
+    command_line = runner_utils.get_param_for_module(block_id, block_dict, 'cmdline')
```
Comment (goravsingal) [inline hubblestack/audit/command_line_parser.py:476]: We are not considering chaining params. As we did in execute(). 
Please check in all modules.

File: hubblestack/audit/command_line_parser.py [hunk 2]
```diff
@@ -457,3 +457,22 @@ def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
         'key_aliases': key_aliases,
         'delimiter': delimiter
     }
+
+
+def get_failure_reason(block_id, block_dict, extra_args=None):
+    """
+
```
Comment (goravsingal) [inline hubblestack/audit/command_line_parser.py:464]: Also please add a one-line about this function (what it does) here.

File: hubblestack/audit/command_line_parser.py [hunk 3]
```diff
@@ -457,3 +457,22 @@ def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
...
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': {'cmdline': 'app --config-file=test test'}, 'status': True},
+                  'caller': 'Audit'}
+    :return:
+    """
+    key_aliases = runner_utils.get_param_for_module(block_id, block_dict, 'key_aliases')
+    command_line = runner_utils.get_param_for_module(block_id, block_dict, 'cmdline')
+    failure_reason = "Fetching value of {0} in {1}".format(key_aliases, command_line)
```
Comment (goravsingal) [inline hubblestack/audit/command_line_parser.py:477]: We can directly return this.

**Review by goravsingal — CHANGES_REQUESTED** [review]
These 3 comments are common for all modules.

File: hubblestack/audit/command_line_parser.py (hunk 1 above)
Comment (MoodyMudit) [reply hubblestack/audit/command_line_parser.py:476]: As discussed, we will do this in future

File: hubblestack/audit/command_line_parser.py (hunk 2 above)
Comment (MoodyMudit) [reply hubblestack/audit/command_line_parser.py]: Done

File: hubblestack/audit/command_line_parser.py (hunk 3 above)
Comment (MoodyMudit) [reply hubblestack/audit/command_line_parser.py]: done

**Review by goravsingal — APPROVED** [review]

**Pull request merged**